podcasts/MainTabBarController.swift

```diff
@@ -442,6 +441,10 @@ class MainTabBarController: UITabBarController, NavigationProtocol {
             self.endOfYear.resetStateIfNeeded()
         }
 
+        NotificationCenter.default.addObserver(forName: .onboardingFlowDidDismiss, object: nil, queue: .main) { notification in
+            self.endOfYear.showPromptBasedOnState(in: self)
```

Comment:
We are missing displaying the red dot on the profile button in the tab bar (ref #393).

I tried adding that here but I'm facing this problem:

<img src="https://user-images.githubusercontent.com/7040243/205908245-9da2158d-b3bd-4737-87f3-34087b110b9c.png" width="300">

For some reason `updateTabBarColor` is not working. 🤔 

This is the code I used:

```swift
            if EndOfYear.isEligible && Settings.showBadgeFor2022EndOfYear {
                self.updateTabBarColor()
                self.profileTabBarItem.badgeValue = "●"
            }
```

Reply:
Oh I managed to fix that. The code using your notification (we might want to extract the badge assignment cause it's the same on `viewDidLoad`:

```swift
        NotificationCenter.default.addObserver(forName: .onboardingFlowDidDismiss, object: nil, queue: .main) { notification in
            self.endOfYear.showPromptBasedOnState(in: self)

            if EndOfYear.isEligible && Settings.showBadgeFor2022EndOfYear {
                self.profileTabBarItem.badgeValue = "●"
            }
        }
```

On `updateTabBarColor` remove the `if EndOfYear.isEligible {` and it works. :)